Wrappers/Python/cil/processors/PaganinProcessor.py

```diff
@@ -0,0 +1,537 @@
+                                                 geometry_parameters[i][1], 
+                                                 parameter))
+                else:
+                    param1 = getattr(geometry, geometry_parameters[i])
+                
+                if (param1 is None) | (param1 == 0):
+                    raise ValueError("Parameter {} cannot be {}, please update\
+                                      data.geometry.{} or over-ride with \
+                                     processor.get_output(override_geometry\
+                                     ={{ '{}' : value }} )"
+                                     .format(parameter, str(param1), 
+                                             str(geometry_parameters[i]),
+                                             parameter))
+                else:
+                    if convert_units[i]:
+                        if unit_multiplier is None:
+                            raise ValueError("Geometry units {} not recognised\
+                                             , expected one of {}, please \
+                                             update data.geometry.config.units"
+                                             .format(str(geometry.config.units)
+                                                     , str(self.UNIT_LIST)))
+                        else:
+                            self.__setattr__(parameter, param1*unit_multiplier)
+                    else:
+                        self.__setattr__(parameter, param1)
+        
+    def _create_filter(self, Nx, Ny, override_filter=None):
+        '''
+        Function to create the Paganin filter, either using the paganin [1] or 
+        generalised paganin [2] method
+        The filter is created on a mesh in Fourier space kx, ky
+        [1] https://doi.org/10.1046/j.1365-2818.2002.01010.x
+        [2] https://iopscience.iop.org/article/10.1088/2040-8986/abbab9 
+        '''
+        if override_filter is None:
+            override_filter = {}
+
+        # update any parameter which has been over-ridden with override_filter
+        if ('alpha' in override_filter) & ('delta' in override_filter):
+            log.warning(msg="Because you specified alpha, it will not be \
+                        calculated and therefore delta will be ignored")
+
+        if ('delta' in override_filter):
+            self.delta = override_filter['delta']
+        else:
+            self.delta = self._delta_user
+        
+        if ('beta' in override_filter):
+            self.beta = override_filter['beta']
+        else:
+            self.beta = self._beta_user
+
+        self._calculate_mu()
+
+        if ('alpha' in override_filter):
+            self.alpha = override_filter['alpha']
+        else:
+            self._calculate_alpha()
+            
+        # create the Fourier mesh
+        kx,ky = np.meshgrid( 
+            np.arange(-Nx/2, Nx/2, 1, dtype=np.float64) 
```

Comment:
Probably the padded size should be a member and not passed in here.

Reply:
Good point, I've updated this